packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -386,3 +544,689 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
+          instance.onTapDown = _handleTrackTapDown;
+        },
+      );
+
+    return gestures;
+  }
+  /// Returns true if the provided [Offset] is located over the track of the
+  /// [RawScrollbar].
+  ///
+  /// Excludes the [RawScrollbar] thumb.
+  @protected
+  bool isPointerOverTrack(Offset position) {
+    if (_scrollbarPainterKey.currentContext == null) {
+      return false;
+    }
+    final Offset localOffset = _getLocalOffset(_scrollbarPainterKey, position);
+    return scrollbarPainter.hitTestInteractive(localOffset)
+      && !scrollbarPainter.hitTestOnlyThumbInteractive(localOffset);
+  }
+  /// Returns true if the provided [Offset] is located over the thumb of the
+  /// [RawScrollbar].
+  @protected
+  bool isPointerOverThumb(Offset position) {
+    if (_scrollbarPainterKey.currentContext == null) {
+      return false;
+    }
+    final Offset localOffset = _getLocalOffset(_scrollbarPainterKey, position);
+    return scrollbarPainter.hitTestOnlyThumbInteractive(localOffset);
+  }
+  /// Returns true if the provided [Offset] is located over the track or thumb
+  /// of the [RawScrollbar].
+  @protected
+  bool isPointerOverScrollbar(Offset position) {
+    if (_scrollbarPainterKey.currentContext == null) {
+      return false;
+    }
+    final Offset localOffset = _getLocalOffset(_scrollbarPainterKey, position);
+    return scrollbarPainter.hitTestInteractive(localOffset);
+  }
+
+  /// Cancels the fade out animation so the scrollbar will remain visible for
+  /// interaction.
+  ///
+  /// Can be overridden by subclasses to respond to a [PointerHoverEvent].
+  ///
+  /// Helper methods [isPointerOverScrollbar], [isPointerOverThumb], and
+  /// [isPointerOverTrack] can be used to determine the location of the pointer
+  /// relative to the painter scrollbar elements.
+  @protected
+  @mustCallSuper
+  void handleHover(PointerHoverEvent event) {
+    // Check if the position of the pointer falls over the painted scrollbar
+    if (isPointerOverScrollbar(event.position)) {
+      _hoverIsActive = true;
+      _fadeoutTimer?.cancel();
+    } else if (_hoverIsActive) {
+      // Pointer is not over painted scrollbar.
+      _hoverIsActive = false;
+      _maybeStartFadeoutTimer();
+    }
+  }
+
+  /// Initiates the fade out animation.
+  ///
+  /// Can be overridden by subclasses to respond to a [PointerExitEvent].
+  @protected
+  @mustCallSuper
+  void handleHoverExit(PointerExitEvent event) {
+    _hoverIsActive = false;
+    _maybeStartFadeoutTimer();
+  }
+
+  @override
+  void dispose() {
+    _fadeoutAnimationController.dispose();
+    _fadeoutTimer?.cancel();
+    scrollbarPainter.dispose();
+    super.dispose();
+  }
+
+
+  @override
+  Widget build(BuildContext context) {
+    updateScrollbarPainter();
+    return NotificationListener<ScrollNotification>(
+      onNotification: _handleScrollNotification,
+      child: RepaintBoundary(
+        child: RawGestureDetector(
+          gestures: _gestures,
```

Comment:
When `ScrollBar` nested, I can not drag the inner thumb, because the gesture event always dispatches to the outer `ScrollBar`, I think we should dispatch the event to the hovered thumb instead:
![20201208_194759](https://user-images.githubusercontent.com/61075224/101480567-d1077c00-398e-11eb-85c4-03e807f14781.gif)

Reply:
@xu-baolin Thank you for the review!! 🙏 
I am curious about this use case. It looks like the Scrollbars are stacked on top of each other. In this case I don't know that the user can distinguish which scrollbar is associated with which scroll view. Since the track area (visible or not) is factored into the hover, both scrollbars are affected. 
In this case I think maybe offsetting the scrollbars, so they align side by side, instead of stacked would probably be a better experience for the user. And would remove the hover issue when they are on top of one another.
Maybe we should document that Scrollbars are not meant to be stacked on top of one another. What do you think?

Reply:
in chrome it's actually just taking priority based on layer hierachy
![image](https://user-images.githubusercontent.com/38903221/101561729-1eeba680-39c6-11eb-948c-eb3211cc95a0.png)

however if the position is not absolute it displays theme side by side 
![image](https://user-images.githubusercontent.com/38903221/101561757-2ca12c00-39c6-11eb-95a1-1ed51fb91f8e.png)